### PR TITLE
[SHELL32] shlexec: Simplify parameters parsing

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Simplify code logic.
JIRA issue: N/A

## Proposed changes

- Use `PathGetArgsW`, `PathRemoveArgsW`, and `PathUnquoteSpacesW` for parsing parameters.

### CODE REVIEW

File: dll/win32/shell32/shlexec.cpp [hunk 1]
```diff
@@ -2169,39 +2169,27 @@ static BOOL SHELL_execute(LPSHELLEXECUTEINFOW sei, SHELL_ExecuteW32 execfunc)
...
+            PathUnquoteSpacesW(wszApplicationName);
+            parametersLen = lstrlenW(pszArgs);
+            if (parametersLen < _countof(parametersBuffer))
             {
-                end = ++src;
-
-                while(isspaceW(*src))
-                    ++src;
+                StringCchCopyW(parametersBuffer, _countof(parametersBuffer), pszArgs);
+                parametersLen = _countof(parametersBuffer);
```
Comment (HBelusca) [inline dll/win32/shell32/shlexec.cpp:2185]: Not sure about that `parametersLen` setting.

Comment (katahiromz) [reply dll/win32/shell32/shlexec.cpp:2185]: This is buffer length that will be used later.

Comment (HBelusca) [reply dll/win32/shell32/shlexec.cpp:2185]: It doesn't appear to be used later.

**Review by HBelusca — APPROVED** [review]
LGTM but let's wait for another review(@whindsaks perhaps?) before merging.

File: dll/win32/shell32/shlexec.cpp (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shell32/shlexec.cpp:2185]:
```suggestion
```

**Pull request merged**